eos-add-flatpak-apps-repos

```diff
@@ -151,6 +151,118 @@ def _update_deploy_file(deploy_file_path, idx, new_val, old_val=None):
         os.chmod(tmp_deploy_file.name, 0o644)
         shutil.copy(tmp_deploy_file.name, deploy_file_path, follow_symlinks=True)
 
+FLATPAKS_TO_MIGRATE = [
+    {
+        'name': 'com.valvesoftware.Steam',
+        'kind': Flatpak.RefKind.APP,
+        'old-branch': 'eos3',
+        'new-branch': 'stable',
+        'old-origin': 'eos-apps',
+        'new-origin': 'flathub'
+    }
+]
+
+def _filter_refs(refs, name=None, prefix=None, kind=None, arch=None,
+                 branch=None, origin=None):
+    ret = []
+    for ref in refs:
+        if kind and kind != ref.get_kind():
+            continue
+        if name and name != ref.get_name():
+            continue
+        if prefix and not ref.get_name().startswith(prefix):
+            continue
+        if arch and arch != ref.get_arch():
+            continue
+        if branch and branch != ref.get_branch():
+            continue
+        if origin and origin != ref.get_origin():
+            continue
+        ret += [ref]
+    return ret
+
+def _migrate_installed_flatpaks():
+    insts = Flatpak.get_system_installations()
+    inst = insts[0]
+
+    refs = inst.list_installed_refs()
+    for migrate in FLATPAKS_TO_MIGRATE:
+        name = migrate['name']
+        kind = migrate['kind']
+        old_branch = migrate.get('old-branch')
+        old_origin = migrate.get('old-origin')
+
+        matching_refs = _filter_refs(refs, name=name, kind=kind,
+                                     branch=old_branch, origin=old_origin)
+        # also search for name.* runtimes to migrate to catch extensions
+        prefix = name + '.'
+        matching_refs += _filter_refs(refs, prefix=prefix,
+                                      kind=Flatpak.RefKind.RUNTIME,
+                                      branch=old_branch, origin=old_origin)
```

Comment:
You could check first if the kind is `RUNTIME`, and in such case you call the function with `prefix=name`. As it will gather the runtime itself + the extensions, and avoids going twice over the list. (Assuming you're not deliberately trying to get runtimes with the same prefix as an app's name)
i.e. only call `_filter_refs` once (though I know we don't have to optimize for speed in such scripts).

Reply:
I didn't do that intentionally because this would widen the scope too much - com.example.runtime would match com.example.runtime2 as well for instance.